test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -500,6 +500,16 @@ def setUp(self):
     devices_job="$devices_base/sbp.service/jobid/$PBS_JOBID"
 else
     devices_job="$devices_base/sbp.slice/sbp-${jobnum}.*.slice"
+fi
+
+device_list=`cat $devices_job/devices.list`
+grep "195" $devices_job/devices.list
+
+ngpus=$(nvidia-smi -L | grep "MIG-GPU" | wc -l)
+if [ "$ngpus" -eq "0" ]; then
+    ngpus=$(nvidia-smi -L | grep "GPU" | wc -l)
```

Comment:
If the call to nvidia-smi -L takes a long time, you could assign the value into a temp file or just a variable so you only need to call it once.

Reply:
It shouldn't take longer than 5 seconds. I don't want to output it to a file, then i'd have to clean it up afterwards.